code/game/machinery/doors/airlock.dm

```diff
@@ -306,6 +306,14 @@ About the new airlock wires panel:
 	else
 		return 0
 
+//Checks if the user gets shocked. Returns TRUE if it happened
+/obj/machinery/door/airlock/proc/shock_user(mob/user, prob)
+	if(!issilicon(user))
```

Comment:
You could compact them into a single statement instead of a nested one. Alternatively, turn them into early returns instead, or group them together by logic (e.g. issilicon and !isElectrified return false, and then return the value of shock(user, prob))